spectree/plugins/flask_plugin.py

```diff
@@ -208,36 +213,56 @@ def validate(
         else:
             model = result
 
-        if resp:
+        if not skip_validation and resp:
             expect_model = resp.find_model(status)
             if resp.expect_list_result(status) and isinstance(model, list):
                 expected_list_item_type = resp.get_expected_list_item_type(status)
                 if all(isinstance(entry, expected_list_item_type) for entry in model):
                     skip_validation = True
                 result = (
                     [
-                        (entry.dict() if isinstance(entry, BaseModel) else entry)
+                        (
+                            serialize_model_instance(entry)
+                            if isinstance(entry, BaseModel)
+                            else entry
+                        )
                         for entry in model
                     ],
                     status,
                     *rest,
                 )
+            elif (
+                expect_model
+                and is_root_model(expect_model)
+                and not isinstance(model, expect_model)
+            ):
+                # Make it possible to return an instance of the model __root__ type
+                # (i.e. not the root model itself).
+                try:
+                    model = expect_model(__root__=model)
+                except ValidationError as err:
+                    resp_validation_error = err
+                else:
+                    skip_validation = True
+                    result = (serialize_model_instance(model), status, *rest)
             elif expect_model and isinstance(model, expect_model):
                 skip_validation = True
-                result = (model.dict(), status, *rest)
+                result = (serialize_model_instance(model), status, *rest)
 
         response = make_response(result)
 
-        if resp and resp.has_model():
+        if resp and resp.has_model() and not resp_validation_error:
```

Comment:
Only do a second validation round if we previously did not find a validation error